### PR TITLE
Fixes #357: Docs: Added error handling section

### LINKED ISSUE #357: Documentation defining good vs error responses
I made a mistake in putting statements in specific command about good vs. exception responses.  Things if successful returns ... or exception.  I think we should make the statement once about the commands that if unsuccessful they return exceptions received from the server or from pywbem interpretation of the response data and let it go at that.  Then remove anything from individual commands about unsuccesful responses except in those cases that differ from this general policy.

### PR DESCRIPTION
**Pull request by andy-maier**
Ready for review and merge.

### CODE REVIEW

File: docs/pywbemclicmdlineinterface.rst [hunk 1]
```diff
@@ -269,3 +269,60 @@ following examples, an underscore ``_`` is shown as the cursor:
...
+    error. This will mostly be caused by CIM errors returned by the server,
+    but can also be caused by the pywbemcli code itself.
+
+* **1 - Python traceback**: In such cases, pywbemcli terminates during its
+  processing, and displays the Python stack traceback on standard error.
+
+  If this happens for a command entered in interactive mode, the pywbemcli shell
+  also terminates with a program exit code of 1.
+
+  These Python tracebacks should never happen and are always considered a
```
Comment (KSchopmeyer) [inline docs/pywbemclicmdlineinterface.rst]: Not completely true.  For an error in a mock-server python script we output the traceback to help the user find their problem.

File: docs/pywbemclicmdlineinterface.rst [hunk 2]
```diff
@@ -269,3 +269,60 @@ following examples, an underscore ``_`` is shown as the cursor:
...
+  `pywbemtools issue tracker <https://github.com/pywbem/pywbemtools/issues>`_`.
+
+* **2 - User error**: In such cases, pywbemcli terminates without even
+  attempting to perform the requested operation, and displays one or more human
+  readable error messages on standard error.
+
+  If this happens for a command entered in interactive mode, the pywbemcli shell
+  is not terminated; only the command that failed is terminated.
+
+  An example for a user error is a missing required argument, or the use of an
```
Comment (KSchopmeyer) [inline docs/pywbemclicmdlineinterface.rst]: or an invalid value on an option

File: docs/pywbemclicmdlineinterface.rst (hunk 1 above)
Comment (andy-maier) [reply docs/pywbemclicmdlineinterface.rst]: I agree. I looked at how it does that and it is an error message with multiple lines that include the traceback. So I'll clarify both the item about errors and the traceback statement you commented on.

BTW, I have opened issue #373 on the way this error is displayed, but I think we can defer that to 0.6.0.

File: docs/pywbemclicmdlineinterface.rst (hunk 2 above)
Comment (andy-maier) [reply docs/pywbemclicmdlineinterface.rst]: DONE

File: docs/pywbemclicmdlineinterface.rst (hunk 1 above)
Comment (andy-maier) [reply docs/pywbemclicmdlineinterface.rst]: DONE

**Pull request merged**